components/cloud/aws/src/util.rs

```diff
@@ -0,0 +1,113 @@
+// Copyright 2020 TiKV Project Authors. Licensed under Apache-2.0.
+
+use std::io::{self, Error, ErrorKind};
+
+use rusoto_core::{
+    region::Region,
+    request::{HttpClient, HttpConfig},
+};
+use rusoto_credential::{
+    AutoRefreshingProvider, AwsCredentials, ChainProvider, CredentialsError, ProvideAwsCredentials,
+};
+use rusoto_sts::WebIdentityProvider;
+
+use async_trait::async_trait;
+
+#[allow(dead_code)] // This will be used soon, please remove the allow.
+const READ_BUF_SIZE: usize = 1024 * 1024 * 2;
+
+pub fn new_http_client() -> io::Result<HttpClient> {
+    let mut http_config = HttpConfig::new();
+    // This can greatly improve performance dealing with payloads greater
+    // than 100MB. See https://github.com/rusoto/rusoto/pull/1227
+    // for more information.
+    http_config.read_buf_size(READ_BUF_SIZE);
+    // It is important to explicitly create the client and not use a global
+    // See https://github.com/tikv/tikv/issues/7236.
+    HttpClient::new_with_config(http_config).map_err(|e| {
+        Error::new(
+            ErrorKind::Other,
+            format!("create aws http client error: {}", e),
+        )
+    })
+}
+
+pub fn get_region(region: &str, endpoint: &str) -> io::Result<Region> {
+    if !endpoint.is_empty() {
+        Ok(Region::Custom {
+            name: region.to_owned(),
+            endpoint: endpoint.to_owned(),
+        })
+    } else if !region.is_empty() {
+        region.parse::<Region>().map_err(|e| {
+            Error::new(
+                ErrorKind::InvalidInput,
+                format!("invalid aws region format {}: {}", region, e),
+            )
+        })
+    } else {
+        Ok(Region::default())
+    }
+}
+
+pub struct CredentialsProvider(AutoRefreshingProvider<DefaultCredentialsProvider>);
+
+impl CredentialsProvider {
+    pub fn new() -> io::Result<CredentialsProvider> {
+        Ok(CredentialsProvider(
+            AutoRefreshingProvider::new(DefaultCredentialsProvider::default()).map_err(|e| {
```

Comment:
why not use `rusoto_core::DefaultCredentialsProvider`

Reply:
The credential provider is a combination of `rusoto_core::DefaultCredentialsProvider` and `WebIdentityProvider`. The later handles IAM in k8s.

Reply:
see above comment that this is the pre-existing rusoto_util code.